framework/sandstone.cpp

```diff
@@ -1039,6 +1041,16 @@ int num_packages() {
     return Topology::topology().packages.size();
 }
 
+void reschedule()
+{
+    if (sApp->device_schedule == nullptr) return;
+    int next_cpu = sApp->device_schedule->get_next_cpu();
+    if (!pin_to_logical_processor(LogicalProcessor(next_cpu))) {
```

Comment:
Either make `get_next_cpu` return a `LogicalProcessor` (and rename to "get_next_processor") or this needs to use `LogicalProcessor(cpu_info[next_cpu].cpu_number)`

I think you need to make it the latter. See below.

Reply:
I prefer the former. The problem using `cpu_info` is that the slices only have the info of the cpus that belong to that specific slice, and I need to reschedule in all available cpus no matter what slice they belong to.

Reply:
I still would like to jump across slices, not sure about jumping across NUMA. But that would be for another upgrade PR.

Reply:
The test that wants to do that could declare itself `isolate_socket`.